src/desktop/mainwindow.cpp

```diff
@@ -1198,6 +1211,87 @@ void MainWindow::onCanvasSaved(const QString &errorMessage)
 		close();
 }
 
+void MainWindow::promptForSaveOnReset(const drawdance::CanvasState &canvasState)
+{
+	const char *objectName = "resetsessionmessagebox";
+	bool saveOrPromptInProgress = m_doc->isSaveInProgress() ||
+		findChild<QMessageBox *>(objectName, Qt::FindDirectChildrenOnly);
+	if(!saveOrPromptInProgress) {
+		QMessageBox::Icon icon;
+		QString message;
+		if(m_doc->isCompatibilityMode()) {
+			icon = QMessageBox::Warning;
+			message = tr(
+				"The session has been reset. Since this is a Drawpile 2.1 "
+				"session and you're running Drawpile 2.2, this probably "
+				"changed how things on the canvas look. Do you want to save "
+				"the canvas as it was before the reset?");
+		} else {
+			icon = QMessageBox::Question;
+			message = tr(
+				"The session has been reset. Normally, everything on the "
+				"canvas should look the same as it did before, but that's "
+				"not guaranteed. Do you want to save the canvas as it was "
+				"before the reset?");
+		}
+
+		QMessageBox *box = new QMessageBox{
+			icon, tr("Session Reset"), message, QMessageBox::Cancel, this};
+		box->setAttribute(Qt::WA_DeleteOnClose);
+		box->setObjectName(objectName);
+		box->setModal(false);
+
+		QPushButton *saveButton = new QPushButton{
+			icon::fromTheme("document-save-as"), tr("Save As..."), box};
+		box->addButton(saveButton, QMessageBox::ActionRole);
+
+		connect(box, &QMessageBox::buttonClicked, this, [=](QAbstractButton *button) {
+			if(button == saveButton) {
+				QString result = FileWrangler{this}.savePreResetImageAs(m_doc, canvasState);
+				if(!result.isEmpty()) {
+					addRecentFile(result);
+				}
+			}
+		});
+
+		box->show();
+	}
+}
+
+void MainWindow::showCompatibilityModeWarning()
+{
+	if(m_doc->isCompatibilityMode()) {
+		QString message = tr(
+			"This session was hosted with Drawpile 2.1. That means several "
+			"Drawpile 2.2 features won't be available, like MyPaint brushes, "
+			"layer folders or the animation timeline. There will also be some "
```

Comment:
I am open to suggestions! I guess it could be trimmed down to something to the effect of "this is a 2.1 session, new stuff will be grayed out", since it'll warn you about desyncs on reset anyway.

Reply:
I think the list of things you mentioned about the blending modes and whatnot are useful to disambiguate. Something like “This session was created by an older version of Drawpile. For compatibility, newer features will be disabled. Due to rendering engine changes, not all users will see the same output, and abrupt changes to the appearance of the drawing after auto-resets is possible.”

Reply:
Will reword to that effect.